### DISCUSSION TITLE
Shorthand for applying multiple modifiers

### DISCUSSION THREAD

**Discussion by Merott** [tailwindlabs/tailwindcss#11236] [Ideas]
This feels like a request/idea that must have come up before, but I couldn't find anything after 10-15 minutes of searching, so here it goes…

Let's say we're styling the hover state of a button (when it's not disabled):

```html
 <button className="bg-blue-500 text-white enabled:hover:-translate-y-px enabled:hover:bg-blue-600 enabled:hover:text-blue-50 enabled:hover:shadow">
```

The `enabled:hover:` modifier repeats for every hover style.

This is a contrived example, but the pattern of repeating modifiers is pretty common, and it's not just pseudo states (e.g. screen size modifiers, sibling states, etc.)

It leads to a lot of repetition, affecting developer productivity and the readability of the code.

I don't know what a good solution could be. I think being able to use syntax like this would be awesome:

```html
<button className="bg-blue-500 text-white enabled:hover:{-translate-y-px bg-blue-600 text-blue-50 shadow}">
```

That said, achieving the above syntax might not be easy (or possible) because it breaks composability, and would also lead to very complex CSS selectors. Perhaps it could be treated as basically syntactic sugar, with a pre-processor to convert it to the existing tailwind syntax?

I'm sure I'm not the first one to think about this. Maybe there are better ways… I'd love to hear some thoughts!

**Comment by wongjn** [comment]
There have been many suggestions about adding such a feature, but [here's one such response from a maintainer](https://github.com/tailwindlabs/tailwindcss/discussions/9053#discussioncomment-3350470):

> We get this question so many times, so I would like to redirect you here for more information: https://twitter.com/adamwathan/status/1461519820411789314

[Also another thread here](https://github.com/tailwindlabs/tailwindcss/discussions/8337#discussioncomment-4032611) with more responses from maintainers.
